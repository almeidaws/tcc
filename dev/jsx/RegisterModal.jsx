import React from 'react';
import Field from './Field.jsx';
import Requisition from './../js/Requisition.js';
import RegisterCredentials from './UserCredentialsField.jsx';
const R = Requisition;

const NameField = props => <Field 
                                type="name"
                                placeholder="Enter Your Name"
                                icon="user"
                                required={true}
                                errorMessage={props.errorMessage}
                                value={props.value}
                                onChange={props.onChange}
                                />
// const EmailField = props => <Field 
//                                 type="email" 
//                                 placeholder="Enter Your Email" 
//                                 icon="envelope" 
//                                 required={true}
//                                 errorMessage={props.errorMessage}
//                                 value={props.value} 
//                                 onChange={props.onChange} />;
// const PasswordField = props => <Field 
//                                     type="password" 
//                                     placeholder="Enter Password" 
//                                     icon="lock" 
//                                     required={true}
//                                     errorMessage={props.errorMessage}
//                                     value={props.value} 
//                                     onChange={props.onChange} />;
const ConfirmPasswordField = props => <Field 
                                type="confirmPassword"
                                placeholder="Please, Repeat Your Password"
                                icon="lock"
                                required={true}
                                errorMessage={props.errorMessage}
                                value={props.value}
                                onChange={props.onChange}
                                />

const RegisterLink = props => <a onClick={props.onClick} href="javascript:{}" className="ms_btn">register now</a>
const LoginLink = props => <p>Already Have An Account? <a href="#myModal1" data-toggle="modal" className="ms_modal hideCurrentModel">login here</a></p>
const ErrorMessage = props => props.error ? <div className="text-warning" role="alert">{props.error}</div> : null;

class RegisterForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = { name: '', email: '', password: '', confirmPassword: ''};
        this.handleName = this.handleName.bind(this);
        // this.handleEmail = this.handleEmail.bind(this);
        // this.handlePassword = this.handlePassword.bind(this);
        this.handleConfirmPassword = this.handleConfirmPassword.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    handleName(event) {
        this.setState({name: event.target.value, error: null});
    }
    // handleEmail(event) { 
    //     this.setState({ email: event.target.value, error: null }); 
    // }
    // handlePassword(event) { 
    //     this.setState({ password: event.target.value, error: null }); 
    // }
    handleConfirmPassword(event) {
        this.setState({ confirmPassword: event.target.value, error: null});
    }
    handleSubmit(event) {
        if (this.validateForm()) {
            // R.register(this.state.name,this.state.email,this.state.password,error => {
            //     if(error === 409) {
            //         this.setState({ error: "Error 409 requisition."})
            //     }
            // })
        }
    }
    validateForm() {
        const form = document.getElementById('registerForm');
        form.classList.add('was-validated');
        return form.checkValidity();
    }
    render() {
        return (
            <form className="ms_register_form" id="registerForm" noValidate>
                <h2>Register</h2>
                {/* <ErrorMessage error={this.state.error} /> */}
                <NameField value={this.state.name} onChange={this.handleName} errorMessage="Invalida name."/>
                {/* <EmailField value={this.state.email} onChange={this.handleEmail} errorMessage="Invalid email. Valid example: john@gmail.com"/>
                <PasswordField value={this.state.password} onChange={this.handlePassword} errorMessage="The password is required"/> */}
                <RegisterCredentials />
                <ConfirmPasswordField value={this.state.confirmPassword} onChange={this.handleConfirmPassword} errorMessage="Password is not equals up field."/>
                <RegisterLink onClick={this.handleSubmit}/>
                <LoginLink />
            </form>
        );
    }
};
const RegisterImage = props => <div className="ms_register_img"> <img src="images/register_img.png" alt="" className="img-fluid" /> </div>;
const CloseButton = props => <button type="button" className="close" data-dismiss="modal"> <i className="fa_icon form_close" /> </button>

class RegisterModal extends React.Component {
    render(){
        return (
            <div id="myModal" className="modal  centered-modal" role="dialog">
                <div className="modal-dialog register_dialog">
                  //Modal Content
                  <div className="modal-content">
                        <CloseButton/>
                        <div className="modal-body">                                    
                            <RegisterImage/>
                            <RegisterForm/>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
};

export default RegisterModal;