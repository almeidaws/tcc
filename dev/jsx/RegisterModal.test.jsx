import React from 'react';
import renderer from 'react-test-renderer';
import RegisterModal from './RegisterModal';

it('<RegisterModal /> should render static content', () => {
    const component = renderer.create(<RegisterModal />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});