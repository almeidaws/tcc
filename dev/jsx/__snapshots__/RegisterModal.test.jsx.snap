// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterModal /> should render static content 1`] = `
<div
  className="modal  centered-modal"
  id="myModal"
  role="dialog"
>
  <div
    className="modal-dialog register_dialog"
  >
    //Modal Content
    <div
      className="modal-content"
    >
      <button
        className="close"
        data-dismiss="modal"
        type="button"
      >
         
        <i
          className="fa_icon form_close"
        />
         
      </button>
      <div
        className="modal-body"
      >
        <div
          className="ms_register_img"
        >
           
          <img
            alt=""
            className="img-fluid"
            src="images/register_img.png"
          />
           
        </div>
        <form
          className="ms_register_form"
          id="registerForm"
          noValidate={true}
        >
          <h2>
            Register
          </h2>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Enter Your Name"
              required={true}
              type="name"
              value=""
            />
            <span
              className="form_icon"
            >
              <i
                aria-hidden="true"
                className="fa_icon form-user"
              />
            </span>
            <div
              className="invalid-feedback text-warning"
            >
              Invalida name.
            </div>
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Enter Your Email"
              required={true}
              type="email"
              value=""
            />
            <span
              className="form_icon"
            >
              <i
                aria-hidden="true"
                className="fa_icon form-envelope"
              />
            </span>
            <div
              className="invalid-feedback text-warning"
            >
              Invalid email. Valid example: john@gmail.com
            </div>
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Enter Password"
              required={true}
              type="password"
              value=""
            />
            <span
              className="form_icon"
            >
              <i
                aria-hidden="true"
                className="fa_icon form-lock"
              />
            </span>
            <div
              className="invalid-feedback text-warning"
            >
              The password is required
            </div>
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Please, Repeat Your Password"
              required={true}
              type="password"
              value=""
            />
            <span
              className="form_icon"
            >
              <i
                aria-hidden="true"
                className="fa_icon form-lock"
              />
            </span>
            <div
              className="invalid-feedback text-warning"
            >
              The password confirmation is required.
            </div>
          </div>
          <a
            className="ms_btn"
            href="javascript:{}"
            onClick={[Function]}
          >
            register now
          </a>
          <p>
            Already Have An Account? 
            <a
              className="ms_modal hideCurrentModel"
              data-toggle="modal"
              href="#myModal1"
            >
              login here
            </a>
          </p>
        </form>
      </div>
    </div>
  </div>
</div>
`;
